### PR TITLE
Rewrite federation user GET/POST/DELETE

### LINKED ISSUE #3791: Rewrite /federations/{{id}}/users to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/federations/{{id}}/users` is handled by Perl

## Expected / new behavior:
`/federations/{{id}}/users` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `GET`
- [ ] `POST`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3791 AND #3792

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control GO Client 
- Traffic Ops

## What is the best way to verify this PR?

Pull down build traffic ops and run API tests. 

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: lib/go-tc/federation.go [hunk 1]
```diff
@@ -103,3 +103,21 @@ type FederationDSPost struct {
...
+	Role     *string `json:"role" db:"role_name"`
+	Username *string `json:"username" db:"username"`
+}
+
+type FederationUserPost struct {
+	IDs     []int `json:"userIds"`
+	Replace *bool `json:"replace"`
+}
+
+func (f *FederationUserPost) Validate(tx *sql.Tx) error {
```
Comment (ocket8888) [inline lib/go-tc/federation.go]: GoDocs on these three entities?

File: traffic_ops/client/federation.go [hunk 2]
```diff
@@ -105,3 +105,41 @@ func (to *Session) DeleteFederationDeliveryService(federationID, deliveryService
 	}
 	return alerts, reqInf, nil
 }
+
+// GetFederationUsers Associates the given Users' IDs to a Federation
+func (to *Session) CreateFederationUsers(federationID int, userIDs []int, replace bool) (ReqInf, error) {
```
Comment (ocket8888) [inline traffic_ops/client/federation.go]: The delete handler returns alerts - do you wanna do that here too?

File: docs/source/api/federations_id_users.rst [hunk 3]
```diff
@@ -37,6 +37,28 @@ Request Structure
...
 	+------+-------------------------------------------------------------------------------------+
 
+.. table:: Request Query Parameters
+
+	+-----------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
+	| Name      | Required | Description                                                                                                                          |
+	+===========+==========+======================================================================================================================================+
+	| userID    | no       | Show only the users that has this integral, unique identifier                                                                        |
+	+-----------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
+	| role      | no       | Show only the users that has this role                                                                                               |
```
Comment (ocket8888) [inline docs/source/api/federations_id_users.rst]: *have

File: docs/source/api/federations_id_users.rst [hunk 4]
```diff
@@ -37,6 +37,28 @@ Request Structure
 	|  ID  | The integral, unique identifier of the federation for which users will be retrieved |
 	+------+-------------------------------------------------------------------------------------+
 
+.. table:: Request Query Parameters
+
+	+-----------+----------+--------------------------------------------------------------------------------------------------------------------------------------+
+	| Name      | Required | Description                                                                                                                          |
+	+===========+==========+======================================================================================================================================+
+	| userID    | no       | Show only the users that has this integral, unique identifier                                                                        |
```
Comment (ocket8888) [inline docs/source/api/federations_id_users.rst]: should be either "the **user** that has this" or "the users that **have** this"

File: lib/go-tc/federation.go (hunk 1 above)
Comment (mhoppa) [reply lib/go-tc/federation.go]: sure!

**Pull request merged**